pyiron_atomistics/vasp/base.py

```diff
@@ -1856,6 +1747,127 @@ def list_potentials(self):
+            if mesh is None:
+                mesh = [int(val) for val in self.input.kpoints[3].split()]
+            self.input.kpoints.set_kpoints_file(
+                size_of_mesh=mesh, shift=center_shift, method="Gamma centered"
+            )
+        if scheme == "GP":
+            self.input.kpoints.set_kpoints_file(
+                size_of_mesh=[1, 1, 1], method="Gamma Point"
+            )
+        if scheme == "Line":
+            if n_path is None and self.input.kpoints._n_path is None:
+                raise ValueError("n_path has to be defined")
+            high_symmetry_points = self.structure.get_high_symmetry_points()
+            if high_symmetry_points is None:
+                raise ValueError("high_symmetry_points has to be defined")
+
+            if path_name is None and self.input.kpoints._path_name is None:
+                raise ValueError("path_name has to be defined")
+            if path_name not in self.structure.get_high_symmetry_path().keys():
+                raise ValueError("path_name is not a valid key of high_symmetry_path")
+
+            if path_name is not None:
+                self.input.kpoints._path_name = path_name
+            if n_path is not None:
+                self.input.kpoints._n_path = n_path
+
+            self.input.kpoints.set_kpoints_file(
+                method="Line",
+                n_path=self.input.kpoints._n_path,
+                path=self._get_path_for_kpoints(self.input.kpoints._path_name),
+            )
+        if scheme == "Manual":
+            if manual_kpoints is None:
+                raise ValueError(
+                    "For the manual mode, the kpoints list should be specified"
+                )
+            else:
+                if weights is not None:
+                    if not (len(manual_kpoints) == len(weights)):
+                        raise AssertionError()
+                self.input.kpoints.set_value(line=1, val=str(len(manual_kpoints)))
+                if reciprocal:
+                    self.input.kpoints.set_value(line=2, val="Reciprocal")
+                else:
+                    self.input.kpoints.set_value(line=2, val="Cartesian")
+                for i, kpt in enumerate(manual_kpoints):
+                    if weights is not None:
+                        wt = weights[i]
+                    else:
+                        wt = 1.0
+                    self.input.kpoints.set_value(
+                        line=3 + i,
+                        val=" ".join([str(kpt[0]), str(kpt[1]), str(kpt[2]), str(wt)]),
+                    )
+
+    def _reinit_potential_setter(self, structure):
+        if structure is not None:
+            self._potential.to_dict().update(
+                {
+                    el: None
+                    for el in set(structure.get_chemical_symbols())
+                    if el not in self._potential.to_dict().keys()
+                }
+            )
+
+    def _get_path_for_kpoints(self, path_name):
```

Comment:
I am just reordering the functions to have the private functions at the end of the class definition.